tutorial/source/index.rst

```diff
@@ -3,92 +3,202 @@
+
+Most users who reach this point will also find our :doc:`guide to tensor shapes in Pyro <tensor_shapes>` essential reading.
+Pyro makes extensive use of the behavior of `"array broadcasting" <https://numpy.org/doc/stable/user/basics.broadcasting.html>`_
+baked into PyTorch and other array libraries to parallelize models and inference algorithms,
+and while it can be difficult to understand this behavior initially, applying the intuition and rules of thumb there
+will go a long way toward making your experience smooth and reducing the prevalence of nasty shape errors.
+
+Core functionality: Deep learning, discrete variables and customizable inference
+---------------------------------------------------------------------------------
+A basic familiarity with this introductory material is all you will need to dive right into using Pyro's two superpowers:
+integration with deep learning and automated exact inference for discrete latent variables.
+The former is described with numerous examples in the series :ref:`deep-generative-models`.
+All are elaborations on the basic idea of the variational autoencoder, introduced in great detail in :doc:`the first tutorial of this series <vae>`.
+
+Pyro's facility with discrete latent variable models like the hidden Markov model is surveyed in the series :ref:`discrete-latent-variables`.
+Making use of this in your own work will require careful reading of :doc:`our overview and programming guide <enumeration>` that opens this series.
+
+Another feature of Pyro is its programmability, the subject of a series of tutorials in :ref:`customizing-inference`.
+Users working with large models where only part of the model needs special attention
+may be interested in `pyro.contrib.easyguide <http://docs.pyro.ai/en/dev/contrib.easyguide.html>`_, introduced in :doc:`the first tutorial of the series <easyguide>`.
+Meanwhile, machine learning researchers interested in developing variational inference algorithms may wish to peruse
+:doc:`the guide to implementing custom variational objectives <custom_objectives>`,
+and a companion example that walks through :doc:`implementing "Boosting BBVI" <boosting_bbvi>`.
+
+Particularly enthusiastic users and potential contributors, especially those interested in contributing to Pyro's core components,
+may even be interested in how Pyro itself works under the hood, partially described in the series :ref:`understanding-pyros-internals`.
+The :doc:`mini-pyro example <minipyro>` contains a complete and heavily commented implementation of a small version of the Pyro language in just a few hundred lines of code,
+and should serve as a more digestable introduction to the real thing.
+
+Tools for specific problems
+-----------------------------
+Pyro is a mature piece of open-source software, not a just science project,
+and in addition to the core machinery for modelling and inference,
+it includes quite a bit of dedicated domain- or problem-specific modelling functionality.
```

Comment:
Let's avoid minimizing the excellent science projects created by many potential Pyro users. Maybe adjust to something like:

> Pyro is a mature probabilistic programming library with "batteries included". In addition to core machinery for modeling and inference, Pyro includes a large toolkit of dedicated ...